src/StableBuffer.mo

```diff
@@ -74,6 +75,25 @@ module {
     };
   };
 
+  /// Removes the item at `index` from the buffer and returns it or `null` if 
+  /// `index >= size`.
+  /// All elements with index > `index` are shifted one position to the left.
+  /// Note: `index` is zero-based.
+  public func remove<X>(buffer: StableBuffer<X>, index : Nat) : ?X {
+    if (index >= buffer.count) {
+      null
+    } else {
+      let element = ?buffer.elems[index];
+      var idx = index;
+      while (idx < (buffer.count - 1 : Nat)) {
+        buffer.elems[idx] := buffer.elems[idx + 1];
+        idx += 1;
+      };
+      buffer.count -= 1;
+      element
+    };
+  };
```

Comment:
You can find the reference implementation here

https://github.com/dfinity/motoko-base/blob/98c5717b283a8ec233907c3a3e92c79e3bbab1b3/src/Buffer.mo#L226

This works, but note there are some slight differences in terms of the additional check on buffer downsizing.

Reply:
Note also that the implementation traps instead of returning null in certain invalid cases.

Reply:
Sure, updated the implementation.

The decision to not implement downsizing was intentional. I noticed that `removeLast` doesn't do it so figured we don't want to do that in `remove` also.